src/DataValues/Time/LocalTime.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace SMW\DataValues\Time;
+
+use DateTime;
+use DateTimeZone;
+use DateInterval;
+use RuntimeException;
+use User;
+
+/**
+ * @license GNU GPL v2+
+ * @since 3.0
+ *
+ * @author mwjames
+ */
+class LocalTime {
+
+	/**
+	 * @see Language::userAdjust
+	 *
+	 * Language::userAdjust cannot be used as entirely relies on the timestamp
+	 * premises making < 1970 return invalid results hence we copy the relevant
+	 * part on work with the DateInterval instead.
+	 *
+	 * @since 3.0
+	 *
+	 * @param DateTime $dateTime
+	 * @param User|null $user
+	 *
+	 * @return DateTime
+	 */
+	public static function adjustTimeByPreference( DateTime $dateTime, User $user = null ) {
+
+		$tz = false;
+
+		if ( $user === null  ) {
```

Comment:
shouldn't this be `if ( $user )`? We did a test with the fix, and now it works instead of using the defaults.